spec/type_references_and_ownership.md

```diff
@@ -0,0 +1,295 @@
+One major advantage of this approach is that the customer cannot really get the link type wrong, because we've autogenerated the expected
+group/kind information all names they supply must point to the right kind of resource. If they are using the `AzureReference` portion
+of `AugmentedKnownTypeReference` we can look up the expected format of the ARM ID for the resource type corresponding to the referenced GVK
+and validate that the customer has provided the correct one.
+
+### A note on the differences between owner references and arbitrary resource references
+Because we are code-generating all of the `ownerRef` fields, we can always supply the annotations for group and kind automatically, using the information
+provided to us by the ARM json specification. **This is not the case for abitrary references (`id`'s) to other resources**. We do not actually know
+programmatically what type that reference is, and it some cases it may actually be allowed to point to multiple different types 
+(think custom image vs shared image gallery).
+
+Currently, the plan is to trust the customer gets it right and ensure we have good
+failure modes for resources if they actually got it wrong (which will cause their resource to move to an unhealthy state). 
+
+## TODOs
+I am planning on adding more details about the below items once I've thought on them more.
+
+1. Expand on what happens when a dependent resource is created and its `ownerRef` points to a resource that doesn't exist.
+2. Same as above, but for a related resource as opposed to a dependent/owner.
+3. Talk about deletion races - if an owning resource is deleted in k8s, Azure deletes (backend) and k8s deletes are all reacing with one another.
+    1. What happens if parent (with same name) is recreated immediately in k8s?
+4. How are CRD YAMLs going to be rendered into actual ARM deployments - are we making one spec with `dependsOn` or sending many different specs?
+   How are we getting ordering right?
+5. How can we allow customers to easily find all dependents for a particular owner (i.e. all subnets of a vnet) using `kubectl`?
+6. Test out an `id` detection heuristic.
+7. How does this property (which doesn't exist in the Azure ARM spec) get transformed into the right ARM JSON?
+
+## Questions
+These are questions I am posing to the group - I don't expect to have an answer without input from the group.
+
+1. Basically everything above is extremely ARM specific. IDs are represented in a different way for data planes and non-ARM endpoints, 
+   and those endpoints JSON schemas may not have the magical `resources` property which allows us to identify owner-dependent relationships.
+   How do we structure the code to solve our ARM problems but allow for alternate solutions for non-ARM?
+2. Is subscription a parameter to in YAMLs, or is the operator configured to act only on a certain subscription?
+   If we only operate on a single sub do we not support cross-sub? Do we care?
```

Comment:
I think starting with a single subscription is probably good enough for a starting point, as long as we don't box ourselves in so that cross subscription is impossible in the future. AFAIK, most Azure customers only use a single subscription (I don't know if this is true for likely ASO customers though)

Reply:
We definitely had a few Batch customers that used multiple subscriptions but not sure how likely this will be in the AKS/ASO world. Would be a good thing to try and validate with others that are more familiar with common usage across AKS customers.

Reply:
Yeah, I think there are some RBAC/auth issues with cross subscription which we would have to navigate, so I agree that as a starting point we probably should focus on single sub (which is what ASO is already doing so we're aligned there)